lang/cs/Org.Apache.REEF.Tang/Exceptions/BindException.cs

```diff
@@ -19,7 +19,7 @@
 
 namespace Org.Apache.REEF.Tang.Exceptions
 {
-    public class BindException : SystemException
+    public class BindException : Exception
```

Comment:
If you change hierarchy of BindException and ClassHierarchyException to inherit directly from Exception, it will change behavior of the whole system, because these exceptions were previously caught as SystemException and now it will be impossible to catch them as ReefRuntimeException. These are not newly introduced exceptions, the existing code relies on them being handled in a certain way, and changing this way might break the system.

Let's not change system behavior as a side effect of a change aimed purely at making code CoreCLR-compatible. If you feel the current exception hierarchy is overcomplicated, please file a separate JIRA for simplifying it and discuss the proposed change there, maybe with the author of the original code.